pkg/ca/fileca/load.go

```diff
@@ -27,21 +27,34 @@ import (
 	"go.step.sm/crypto/pemutil"
 )
 
-func loadKeyPair(certPath, keyPath, keyPass string) (*x509.Certificate, crypto.Signer, error) {
+func loadKeyPair(certPath, keyPath, keyPass string) ([]*x509.Certificate, crypto.Signer, error) {
 
 	var (
-		cert *x509.Certificate
-		err  error
-		key  crypto.Signer
+		certs []*x509.Certificate
+		err   error
+		key   crypto.Signer
 	)
 
-	// TODO: Load chain of certs (intermediates and root) instead of just one
-	// certificate.
-	cert, err = pemutil.ReadCertificate(certPath)
+	certs, err = pemutil.ReadCertificateBundle(certPath)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	// Verify certificate chain
+	{
```

Comment:
This block of code treats all certificates in the bundle as root certificates, trying to build a valid chain from certs[0] to any other cert in the root cert pool.

Assuming the order of certs is intermediate1 (that will be used to issue leaf certs), i2, i3...root, you'll want to create a second cert pool for intermediates and use the Intermediates option from [VerifyOptions](https://pkg.go.dev/crypto/x509#VerifyOptions). So something like: 
* Add certs[last index] to root pool
* Add certs[1 .. last index - 1] to intermediate pool
* Verify certs[0]

Reply:
One other small thing, VerifyOptions will need to include the extended key usages that are used in all certificates. 

```
&VerifyOptions {
  ...
  KeyUsages: []x509.ExtKeyUsage{
    x509.ExtKeyUsageCodeSigning,
  },
}
```

Reply:
Go's x509 library won't enforce that CA certificates contain the same EKUs, but some certificate verification libraries will. For Go, it'll just skip over any cert in the chain that contains no EKUs - https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/crypto/x509/verify.go;l=1065-1067;drc=refs%2Ftags%2Fgo1.17.6. At a minimum, you could include the code signing EKU in just the leaf cert and include the EKU in `VerifyOptions` and everything would work.

For some background, it's called EKU chaining, where all intermediate certificates need to have the EKUs of the certificates they issue (root certs don't require EKUs). The idea is it restricts the types of certs a certificate authority can issue. This comes up with web PKI, as it's a requirement from the CA/Browser (CAB) forum (https://cabforum.org/wp-content/uploads/CA-Browser-Forum-BR-1.8.0.pdf). See *7.1.5 Name constraints*:
> For a Subordinate CA Certificate to be considered Technically Constrained, the certificate MUST include an Extended Key Usage (EKU) extension specifying all extended key usages that the Subordinate CA Certificate is authorized to issue certificates for. The anyExtendedKeyUsage KeyPurposeId MUST NOT appear within
this extension."

FWIW, this is still useful with private PKI, to limit what an intermediate CA can issue, but technically Go won't enforce this. 

Totally unrelated to this PR, in case you're curious, there was an issue [last year](https://groups.google.com/g/mozilla.dev.security.policy/c/EzjIkNGfVEE/m/XSfw4tZPBwAJ) ([another explanation](https://www.hardenize.com/blog/revocation-of-misissued-ocsp-signing-intermediates)) that was related to EKU chaining where intermediate CAs were being incorrectly issued with the EKU for OCSP because they thought it was necessary for chaining.

Reply:
Amazing!!! Thanks so much for this detailed explanation!